temperature/crud.py

```diff
@@ -0,0 +1,55 @@
+from datetime import datetime
+
+import httpx
+from fastapi import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+from . import models
+
+
+async def fetch_current_temperature(city_name: str) -> float:
+    """
+    Fetches current temperature for a given city from MetaWeather API.
+    """
+    # Запрос для получения WOEID города
+    location_url = f"https://www.metaweather.com/api/location/search/?query={city_name}"
+    async with httpx.AsyncClient() as client:
+        location_response = await client.get(location_url)
+        if location_response.status_code != 200 or not location_response.json():
+            raise HTTPException(status_code=404, detail=f"City '{city_name}' not found.")
+
+        location_data = location_response.json()
+        woeid = location_data[0]["woeid"]
+
+        # Запрос для получения погоды по WOEID
+        weather_url = f"https://www.metaweather.com/api/location/{woeid}/"
+        weather_response = await client.get(weather_url)
+        if weather_response.status_code != 200:
+            raise HTTPException(status_code=404, detail=f"Weather data for '{city_name}' not found.")
```

Comment:
Ensure that the error handling for the weather data request is comprehensive, potentially including logging for debugging purposes.